### ISSUE TITLE
Some cache groups should be global on multisite

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/wordpoints#275]
We currently have the following cache groups:

```
wordpoints_modules
wordpoints_network_points_logs_query
wordpoints_points_logs_query
wordpoints_points_top_users
wordpoints_user_ranks
wordpoints_ranks
wordpoitns_users_with_rank
```
Only `wordpoints_network_points_logs_query` is global on multisite.

However, when WordPoints is network active, the `wordpoints_points_top_users` group should be global.

I think that the `wordpoints_ranks`, `wordpoints_user_ranks`, and `wordpoints_users_with_rank` groups should be global in that case as well. However, I'm not 100% certain that we have actually implemented ranks to be network-wide when WordPoints is network active. 

I'm not sure if the intention was to have ranks be network-wide when network-active, or even if that would be the best way to implement it. It will take some thought and investigation, which will result in a spin-off ticket. Perhaps it is time to let components be activated per-site/network-wide when WordPoints is network-active on multisite.

**Comment by JDGrimes** [comment]
The call to `wp_cache_add_global_groups()` should be inside a `function_exists()` check, because the function wasn't introduced until [WP@7986](https://core.trac.wordpress.org/changeset/7986), that is, not until after the rest of the cache system was introduced. This means that it might not be available in some really old cache drop-ins. For this reason, it is always used with a conditional check in WordPress core.

**Comment by JDGrimes** [comment]
Rank groups, and therefore the ranks themselves, are not network-wide even when WordPoints is network-active. This should possibly change, but that belongs in another ticket.

**Comment by JDGrimes** [comment]
One cache group that I hadn't considered was the points logs meta cache, because it is handled by the `*_metadata()` functions. The logs are all stored in a single table on multisite, regardless of how WordPoints is activated. This means that they have unique IDs within a network. It also means that we can run network points log queries, which are supported by the query class (and are cached in a global cache group). This could cause potential issues, however, because if a network query is performed and the metadata is used in some way the metadata from other site's points logs will be loaded into the cache. This causes cache duplication, which isn't necessarily a huge issue. However, it becomes an issue when metadata later gets deleted, because the only the current site's cache is cleaned. This could then lead to the cache being stale on other sites, which would inevitably cause odd behavior.

Because of this, the `wordpoints_points_log_meta` group should be network-wide (or "global") as well.
